docs/appendices/release-notes/5.9.0.rst

```diff
@@ -59,6 +59,15 @@ Breaking Changes
   regressions on tables created before 5.9.0. For tables created on and after
   5.9.0, the fix has positive impact on the performance.
 
+- Fixed an issue that caused ``WHERE`` clause containing ``NOT`` operator on
+  an array type against an empty array to incorrectly filter array of nulls. It
+  is a breaking change because the fix causes performance degradations.
+
+- Fixed an issue that caused ``WHERE`` clause containing ``NOT`` operator on
+  an array type without doc-values against a non-empty array to incorrectly
+  un-filter null rows. It is a breaking change because the fix causes
+  performance degradations.
+
```

Comment:
2 questions:

1. Do I get it right, that this PR also fixes https://github.com/crate/crate/issues/16476? 
I guess we don't want to backport because it's a breaking change and not a bug fix?

2. Shouldn't we phrase it similar to https://github.com/crate/crate/commit/6c63ebf5feee7ef8db1a6a3158e416d98053166d
ie add
  ...```
regressions on tables created before 5.9.0. For tables created on and after
  5.9.0, the fix has positive impact on the performance
```

to both entries?

Also, could you please measure the impact of the second commit, ie compare it with numbers you [posted](https://github.com/crate/crate/pull/16427#issuecomment-2257020457) for the first commit?

Reply:
> Do I get it right, that this PR also fixes https://github.com/crate/crate/issues/16476?
I guess we don't want to backport because it's a breaking change and not a bug fix?

Yes, we should not backport this.

> Shouldn't we phrase it similar to https://github.com/crate/crate/commit/6c63ebf5feee7ef8db1a6a3158e416d98053166d

Thanks, added